src/proxy.rs

```diff
@@ -483,9 +483,19 @@ pub async fn freebind_connect(
 // as a best guess.
 pub fn guess_inbound_service(
     conn: &Connection,
+    for_host_header: &Option<String>,
     upstream_service: Vec<Service>,
     dest: &Workload,
 ) -> Option<ServiceDescription> {
+    // First, if the client told us what Service they were reaching, look for that
+    // Note: the set of Services we look for is bounded, so we won't blindly trust bogus info.
```

Comment:
```suggestion
    // Note: the set of Services we look for is bounded, so we won't blindly trust bogus info.
```

Bounded by what?

Reply:
From XDS. The problem is there are N services for 1 pod, so we don't know which one. They tell us a arbirtry service, and we use that IFF its in the list of services for the workload